packages/core-utils/src/itinerary.ts

```diff
@@ -681,3 +681,22 @@ export const convertGraphQLResponseToLegacy = (leg: any): any => ({
   tripHeadsign: leg.trip?.tripHeadsign,
   tripId: leg.trip?.gtfsId
 });
+
+/** Extracts the route number for a leg returned from OTP1 or OTP2. */
+export const getLegRouteShortName = (
+  leg: Pick<Leg, "route" | "routeShortName">
+): string => {
+  const { route, routeShortName } = leg;
+  if (typeof route === "string") {
+    return typeof routeShortName === "string" ? routeShortName : route;
+  }
+  return route?.shortName;
+};
+
+/** Extract the route ling name for a leg returned from OTP1 or OTP2. */
```

Comment:
Typo: `ling name`

Reply:
Thanks, fixed in https://github.com/opentripplanner/otp-ui/pull/634/commits/1b3839f259632ce90dad68b5149df3258e54fe4e.